spec/features/user_post_index_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+require 'capybara/rspec'
+
+RSpec.describe 'User Post Index Page', type: :feature do
+  before do
+    # Set up any necessary data or visit relevant pages
+  end
+
+  it 'displays the user profile picture' do
+    # Test code goes here
+  end
+
+  it 'displays the user username' do
+    # Test code goes here
+  end
+
+  it 'displays the number of posts the user has written' do
+    # Test code goes here
+  end
+
+  it 'displays a post title' do
+    # Test code goes here
+  end
+
+  it 'displays part of the post body' do
+    # Test code goes here
+  end
+
+  it 'displays the first comments on a post' do
+    # Test code goes here
+  end
+
+  it 'displays the number of comments a post has' do
+    # Test code goes here
+  end
+
+  it 'displays the number of likes a post has' do
+    # Test code goes here
+  end
```

Comment:
- [ ] Good to have this user post index test file. :+1:  But please from the project requirement, you should write a test for the following :point_down: But now you have no test code in your `it` block to test the following:blush: 
  
  * Tests that asserts you can see the user's profile picture.
  * Tests that asserts you can see the user's username.
  * Tests that asserts you can see the number of posts the user has written.
  * Tests that asserts you can see a post's title.
  * Tests that asserts you can see some of the post's body.
  * Tests that asserts you can see the first comments on a post.
  * Tests that asserts you can see how many comments a post has.
  * Tests that asserts you can see how many likes a post has.
  * Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
  * Tests that redirects to the post's show page when you click on a post.